internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -423,20 +364,56 @@ func readPlainTextSecret(opts *CreateOpts) error {
 	return nil
 }
 
-func readConfigFile(opts *CreateOpts) (SecretConfig, error) {
-	var sc SecretConfig
+func readConfigFile(opts *CreateOpts) (SecretConfig, detailsInternal, error) {
+	var (
+		sc SecretConfig
+		di detailsInternal
+	)
 
-	f, err := os.ReadFile(opts.SecretFilePath)
-	if err != nil {
-		return sc, fmt.Errorf("unable to open config file: %w", err)
+	if err := hclsimple.DecodeFile(opts.SecretFilePath, nil, &sc); err != nil {
+		return sc, di, fmt.Errorf("failed to decode config file: %w", err)
 	}
 
-	err = yaml.Unmarshal(f, &sc)
+	detailsMap, err := ctyValueToMap(sc.Details)
 	if err != nil {
-		return sc, fmt.Errorf("unable to unmarshal config file: %w", err)
+		return sc, di, err
+	}
+	di.Details = detailsMap
+
+	return sc, di, nil
+}
+
+func ctyValueToMap(value cty.Value) (map[string]any, error) {
+	varMapNow := make(map[string]any)
+	for k, v := range value.AsValueMap() {
+		if v.Type() == cty.String {
+			if k == "rotation_policy_name" {
+				varMapNow[k] = rotationPolicies[v.AsString()]
```

Comment:
I think it may be better to accept full policy name here (e.g. "built-in:30-days-2-active") instead of just "30". This will allow us to support custom policy names in the future and it would be consistent with our current API.